backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchBackendImpl.java

```diff
@@ -220,38 +204,8 @@ public IndexManagerBuilder createIndexManagerBuilder(
 		);
 	}
 
-	private IndexNames createIndexNames(EventContext indexEventContext, String hibernateSearchIndexName,
-			String mappedTypeName) {
-		IndexLayoutStrategy indexLayoutStrategy = indexLayoutStrategyHolder.get();
-		URLEncodedString writeAlias = IndexNames.encodeName( indexLayoutStrategy.createWriteAlias( hibernateSearchIndexName ) );
-		URLEncodedString readAlias = IndexNames.encodeName( indexLayoutStrategy.createReadAlias( hibernateSearchIndexName ) );
-
-		URLEncodedString primaryName = null;
-		if ( writeAlias == null || readAlias == null ) {
-			primaryName = IndexNames.encodeName(
-					indexLayoutStrategy.createInitialElasticsearchIndexName( hibernateSearchIndexName ) );
-		}
-		else if ( writeAlias.equals( readAlias ) ) {
-			throw log.sameWriteAndReadAliases( writeAlias, indexEventContext );
-		}
-
-		URLEncodedString readName = readAlias != null ? readAlias : primaryName;
-		URLEncodedString writeName = writeAlias != null ? writeAlias : primaryName;
-
-		IndexNames indexNames = new IndexNames( hibernateSearchIndexName,
-				writeName, writeAlias != null, readName, readAlias != null );
-
-		// This will check that names are unique.
-		indexNamesRegistry.register( indexNames );
-
-		// This will allow the type mapping to resolve the type name from the index name.
-		typeNameMapping.register( indexNames, mappedTypeName );
-
-		return indexNames;
-	}
-
```

Comment:
This could have remained here, since you expose the typeNameMapping and layout strategy from the ElasticsearchLink anyway. But ok.